pkg/kubevirt/data_volume.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubevirt
+
+import (
+	"context"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cdi "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// ClientFactory creates a client from the kubeconfig saved in the "kubeconfig" field of the given secret.
+type ClientFactory interface {
+	// GetClient creates a client from the kubeconfig saved in the "kubeconfig" field of the given secret.
+	// It also returns the namespace of the kubeconfig's current context.
+	GetClient(kubeconfig []byte) (client.Client, string, error)
+}
+
+// ClientFactoryFunc is a function that implements ClientFactory.
+type ClientFactoryFunc func(kubeconfig []byte) (client.Client, string, error)
+
+// GetClient creates a client from the kubeconfig saved in the "kubeconfig" field of the given secret.
+// It also returns the namespace of the kubeconfig's current context.
+func (f ClientFactoryFunc) GetClient(kubeconfig []byte) (client.Client, string, error) {
+	return f(kubeconfig)
+}
+
+// DataVolumeManager manages Kubevirt DataVolume operations.
+type DataVolumeManager interface {
+	// CreateOrUpdateVolume creates a new kubevirt Data Volume from the data volume specs and in the passed namespace.
+	CreateOrUpdateVolume(ctx context.Context, kubeconfig []byte, dataVolume *cdi.DataVolume) error
```

Comment:
Could you please rename this method to `CreateOrUpdateDataVolume`, and similarly rename the other methods of this interface? I think this would enhance clarity and consistency.

Reply:
Yeah, that makes total sense, will adjust it accordingly.